gui/gui/src/TGFileDialog.cxx

```diff
@@ -657,6 +684,9 @@ Bool_t TGFileDialog::ProcessMessage(Long_t msg, Long_t parm1, Long_t)
          break;
 
       case kC_TEXTENTRY:
+        if((fDlgType==kDOpen || fDlgType==kDSave) && fOk->GetState()==kButtonDisabled)//when typing, re-enable previously disabled button after having clicked on file instead of folder
```

Comment:
```suggestion
        if ((fDlgType==kDOpen || fDlgType==kDSave) && fOk->GetState()==kButtonDisabled) //when typing, re-enable previously disabled button after having clicked on file instead of folder
```